src/Alluxio.cc

```diff
@@ -0,0 +1,659 @@
+{
+    Env env;
+    jvalue jFileOptions;
+
+    // First get the ClientContext.Configuration
+    env.callStaticMethod(&jFileOptions, 
+            "alluxio/client/file/options/OpenFileOptions", "defaults", 
+            "()Lalluxio/client/file/options/OpenFileOptions;");
+
+    return new AlluxioOpenFileOptions(env, jFileOptions.l);
+}
+
+void AlluxioOpenFileOptions::setReadType(ReadType readType)
+{
+    jobject jReadType;
+    jvalue  ret;
+
+    // Get the enum value for the specified write type
+    jReadType = enumObjReadType(m_env, readType);
+    
+    // Call the method to set the write type
+    m_env.callMethod(&ret, m_obj, "setReadType", 
+            "(Lalluxio/client/ReadType;)Lalluxio/client/file/options/OpenFileOptions;",
+            jReadType);
+}
+
+jAlluxioFileSystem AlluxioFileSystem::getFileSystem(AlluxioClientContext *acc)
+{
+  Env env;
+  jvalue ret;
+
+  // might throw exception, caller needs to handle it
+  env.callStaticMethod(&ret, "alluxio/client/file/BaseFileSystem", "get", 
+                "()Lalluxio/client/file/BaseFileSystem;");
+  
+  return new AlluxioFileSystem(env, ret.l, acc);
+}
+
+jAlluxioFileSystem AlluxioFileSystem::copyClient(jAlluxioFileSystem client)
+{
+  return new AlluxioFileSystem(client->getEnv(), client->getJObj(), 
+          client->clientContext);
+}
+
+
+void AlluxioFileSystem::deletePath(const char * path)
+{
+  jvalue ret;
+
+  jAlluxioURI uri = AlluxioURI::newURI(path);
+
+  m_env.callMethod(&ret, m_obj, "delete", "(Lalluxio/AlluxioURI;)V",
+                   uri->getJObj());
+
+  delete uri;
+  return;
+}
+
+bool AlluxioFileSystem::exists(const char * path)
+{
+  jvalue ret;
+
+  jAlluxioURI uri = AlluxioURI::newURI(path);
+
+  m_env.callMethod(&ret, m_obj, "exists", "(Lalluxio/AlluxioURI;)Z",
+                   uri->getJObj());
+
+  delete uri;
+  return ret.z;
+}
+
+jFileOutStream AlluxioFileSystem::createFile(const char * path, 
+        AlluxioCreateFileOptions *options)
+{
+    jvalue ret;
+
+    jAlluxioURI uri = AlluxioURI::newURI(path);
+
+    jFileOutStream fileOutStream = NULL;
+
+    if (uri == NULL)
+    {
+       goto exit;
+    }
+
+    m_env.callMethod(&ret, m_obj, "createFile", 
+            "(Lalluxio/AlluxioURI;Lalluxio/client/file/options/CreateFileOptions;)Lalluxio/client/file/FileOutStream;",
+            uri->getJObj(), options->getOptions());
+
+    delete uri;
+
+    // FIXME: Change to shared_ptr
+    fileOutStream = new FileOutStream(m_env, ret.l); 
+
+exit:
+    return fileOutStream;
+}
+
+jFileOutStream AlluxioFileSystem::createFile(const char * path)
+{
+    jvalue ret;
+
+    jAlluxioURI uri = AlluxioURI::newURI(path);
+
+    jFileOutStream fileOutStream = NULL;
+
+    if (uri == NULL)
+    {
+       goto exit;
+    }
+
+    m_env.callMethod(&ret, m_obj, "createFile", 
+            "(Lalluxio/AlluxioURI;)Lalluxio/client/file/FileOutStream;",
+            uri->getJObj());
+
+    delete uri;
+
+    // FIXME: Change to shared_ptr
```

Comment:
This is a good point. The current APIs all assume callers will de-allocate the returned objects. We should consider converting all jXXX type into a smart pointer. Maybe unique_ptr?